packages/zoltan2/test/color/coloring1.cpp

```diff
@@ -258,29 +382,45 @@ int main(int narg, char** arg)
     }
     colorFile.close();
   }
-
+  
   // Print # of colors on each proc.
-  std::cout << "No. of colors on proc " << me << " : " << soln->getNumColors() << std::endl;
-
+  //if(checkLength>0) std::cout << "No. of colors on proc " << me << " : " << soln->getNumColors() << std::endl;
+  if(checkLength>0){
+    localColors = soln->getNumColors();
+  }
+  
+  //colors are guaranteed to overlap, max will discard duplicates
+  Teuchos::reduceAll<int,int>(*comm, Teuchos::REDUCE_MAX,1,&localColors,&totalColors);
+  
   std::cout << "Going to validate the soln" << std::endl;
   // Verify that checkColoring is a coloring
-  testReturn = validateColoring(Matrix, checkColoring);
-
-  // Check balance (not part of pass/fail for now)
-  checkBalance((zlno_t)checkLength, checkColoring);
-
+  if(colorAlg=="2GL" ||colorAlg == "Hybrid"){
+    //need to check a distributed coloring
+    testReturn = validateDistributedColoring(Matrix, checkColoring, me);
+  } else if (checkLength > 0){
+    testReturn = validateColoring(Matrix, checkColoring);
+  }
+    // Check balance (not part of pass/fail for now)
+  if(checkLength > 0) checkBalance((zlno_t)checkLength, checkColoring);
+  
   } catch (std::exception &e){
       std::cout << "Exception caught in coloring" << std::endl;
       std::cout << e.what() << std::endl;
       std::cout << "FAIL" << std::endl;
       return 0;
   }
-
+  int numGlobalConflicts = 0;
+  Teuchos::reduceAll<int,int>(*comm, Teuchos::REDUCE_MAX,1,&testReturn,&numGlobalConflicts);
```

Comment:
This overload may have been deprecated.  Use this instead:
```cpp
Teuchos::reduceAll(*comm, Teuchos::REDUCE_MAX, testReturn, Teuchos::outArg(numGlobalConflicts));
```

Reply:
I've been told that this call matches the usage across Zoltan2, and it is not deprecated.